toucan_connectors/trello/trello_connector.py

```diff
@@ -0,0 +1,148 @@
+            'token': self.token, **customParams
+        }).json()
+
+    def replace_id_by_value(self, card_with_id, lists=None, labels=None,
+                            members=None, custom_fields=None):
+        """
+        `card_with_id` is a dictionary containing all data of a card,
+        but with unreadlable id instead of value
+        This fonction return `card_with_value` with the same dictionnary
+        as `card_with_id` but with readable value
+
+        `lists`: dictionnary of correspondance between list names and ids
+        `members`: dictionnary of correspondance between members names and ids
+        `custom_fields`: dictionnary of correspondance between custom field and there representation
+        """
+        card_with_value = {}
+
+        # id, name and url fields do not need to translate from id to value
+        card_with_value["id"] = card_with_id["id"]
+
+        if 'name' in card_with_id.keys():
+            card_with_value['name'] = card_with_id['name']
+        if 'url' in card_with_id.keys():
+            card_with_value['url'] = card_with_id['url']
+
+        # lists, members and labels need to translate from a id to a value
+        if lists:
+            card_with_value['lists'] = lists[card_with_id['idList']]
+        if members:
+            card_with_value['members'] = [members[member] for member in card_with_id['idMembers']]
+        if labels:
+            card_with_value['labels'] = [labels[label['id']] for label in card_with_id['labels']]
+
+        # custom fields
+        if custom_fields:
+            for card_custom_field in card_with_id['customFieldItems']:
+                custom_field = custom_fields[card_custom_field['idCustomField']]
+                if custom_field['type'] == 'number':
+                    card_with_value[custom_field['name']] = float(
+                        card_custom_field['value']['number'])
+                elif custom_field['type'] == 'text':
+                    card_with_value[custom_field['name']] = card_custom_field['value']['text']
+                elif custom_field['type'] == 'date':
+                    card_with_value[custom_field['name']] = card_custom_field['value']['date']
+                elif custom_field['type'] == 'checkbox':
+                    if card_custom_field['value']['checked'] == 'true':
+                        card_with_value[custom_field['name']] = True
+                elif custom_field['type'] == 'list':
+                    list_of_options = {x["id"]: x for x in custom_field['options']}
+                    option = list_of_options[card_custom_field['idValue']]
+                    card_with_value[custom_field['name']] = option['value']['text']
+                else:
+                    warnings.warn(f"""The custom field {custom_field['name']} used into \
+                    your board is of type {custom_field['type']}.
+                    This type is not handled by this app and is simply ignored.""")
```

Comment:
I think this code block from 82:100 makes the method too long : 

- at least it should be put in a separate function that deals with custom fields
- could be simplified with a mapping type -> function, I can help with this